File: src/main/dart/net/coderextreme/data/CADPartChildNoTransformation.dart
// Copyright 2021 by John Carlson.  All rights reserved.
import 'dart:io';
import '../x3d.dart';
var X3D0 =  X3D(

      profile_ : SFString('CADInterchange'),
      version_ : SFString('4.0'),
      head_ : 
        head(
          meta_ : [
            meta(
              name_ : SFString('title'),
              content_ : SFString('CADPartChildNoTransformation.x3d')),

            meta(
              name_ : SFString('description'),
              content_ : SFString('Developer scene to explore Mantis Issue 528, providing a baseline example with no transformation of CADPart children.')),

            meta(
              name_ : SFString('creator'),
              content_ : SFString('Vince Marchetti, Don Brutzman, Hyokwang Lee, Dick Puk')),

            meta(
              name_ : SFString('translator'),
              content_ : SFString('Vince Marchetti bushing generation using python scripts')),

            meta(
              name_ : SFString('reference'),
              content_ : SFString('https://www.web3d.org/member-only/mantis/view.php?id=528')),

            meta(
              name_ : SFString('warning'),
              content_ : SFString('This scene is intended for specification development only.')),

            meta(
              name_ : SFString('created'),
              content_ : SFString('29 June 2012')),

            meta(
              name_ : SFString('modified'),
              content_ : SFString('20 October 2019')),

            meta(
              name_ : SFString('identifier'),
              content_ : SFString('https://www.web3d.org/x3d/content/examples/Basic/CAD/CADPartChildNoTransformation.x3d')),

            meta(
              name_ : SFString('generator'),
              content_ : SFString('X3D-Edit 3.3, https://savage.nps.edu/X3D-Edit')),
          component_ : 
            component(
              name_ : SFString('CADGeometry'),
              level_ : 2)],
          component_ : 
            component(
              name_ : SFString('EnvironmentalEffects'),
              level_ : 1)),
      Scene_ : 
        Scene(
          children_ : [
            WorldInfo(
              title_ : SFString('CADPartChildNoTransformation.x3d')),

            Background(
              skyColor_ : MFColor([SFColor(0.9607843), SFColor(1), SFColor(0.9607843)])),

            Viewpoint(
              description_ : SFString('Hello CAD bushing'),
              position_ : SFVec3f([SFDouble(0), SFDouble(5), SFDouble(15)]),
              orientation_ : SFRotation([SFDouble(1), SFDouble(0), SFDouble(0), SFDouble(-0.321751)])),

            CADAssembly(
              name_ : SFString('DesignPatternAssembly'),
              children_ : [
                CADPart(
                  name_ : SFString('CADPartExample'),
                  children_ : [
                    CADFace(
                      name_ : SFString('outerSurface'),
                      shape_ : 
                        Shape(
                          appearance_ : 
                            Appearance(
                              DEF_ : SFString('AppearanceGrey'),
                              material_ : 
                                Material()),
                          geometry_ : 
                            Cylinder(
                              top_ : false,
                              bottom_ : false,
                              height_ : 3,
                              radius_ : 2))),

                    CADFace(
                      name_ : SFString('innerSurface'),
                      shape_ : 
                        Shape(
                          appearance_ : 
                            Appearance(
                              USE_ : SFString('AppearanceGrey')),
                          geometry_ : 
                            Cylinder(
                              top_ : false,
                              bottom_ : false,
                              height_ : 3,
                              solid_ : false))),

                    CADFace(
                      name_ : SFString('topCap'),
                      shape_ : 
                        Shape(
                          appearance_ : 
                            Appearance(
                              USE_ : SFString('AppearanceGrey')),
                          geometry_ : 
                            IndexedTriangleSet(
                              index_ : MFInt32([SFInt32(0), SFInt32(1), SFInt32(2), SFInt32(2), SFInt32(1), SFInt32(3), SFInt32(2), SFInt32(3), SFInt32(4), SFInt32(4), SFInt32(3), SFInt32(5), SFInt32(4), SFInt32(5), SFInt32(6), SFInt32(6), SFInt32(5), SFInt32(7), SFInt32(6), SFInt32(7), SFInt32(8), SFInt32(8), SFInt32(7), SFInt32(9), SFInt32(8), SFInt32(9), SFInt32(10), SFInt32(10), SFInt32(9), SFInt32(11), SFInt32(10), SFInt32(11), SFInt32(12), SFInt32(12), SFInt32(11), SFInt32(13), SFInt32(12), SFInt32(13), SFInt32(14), SFInt32(14), SFInt32(13), SFInt32(15), SFInt32(14), SFInt32(15), SFInt32(16), SFInt32(16), SFInt32(15), SFInt32(17), SFInt32(16), SFInt32(17), SFInt32(18), SFInt32(18), SFInt32(17), SFInt32(19), SFInt32(18), SFInt32(19), SFInt32(20), SFInt32(20), SFInt32(19), SFInt32(21), SFInt32(20), SFInt32(21), SFInt32(22), SFInt32(22), SFInt32(21), SFInt32(23), SFInt32(22), SFInt32(23), SFInt32(24), SFInt32(24), SFInt32(23), SFInt32(25), SFInt32(24), SFInt32(25), SFInt32(26), SFInt32(26), SFInt32(25), SFInt32(27), SFInt32(26), SFInt32(27), SFInt32(28), SFInt32(28), SFInt32(27), SFInt32(29), SFInt32(28), SFInt32(29), SFInt32(30), SFInt32(30), SFInt32(29), SFInt32(31), SFInt32(30), SFInt32(31), SFInt32(32), SFInt32(32), SFInt32(31), SFInt32(33), SFInt32(32), SFInt32(33), SFInt32(34), SFInt32(34), SFInt32(33), SFInt32(35), SFInt32(34), SFInt32(35), SFInt32(36), SFInt32(36), SFInt32(35), SFInt32(37), SFInt32(36), SFInt32(37), SFInt32(38), SFInt32(38), SFInt32(37), SFInt32(39), SFInt32(38), SFInt32(39), SFInt32(40), SFInt32(40), SFInt32(39), SFInt32(41), SFInt32(40), SFInt32(41), SFInt32(42), SFInt32(42), SFInt32(41), SFInt32(43), SFInt32(42), SFInt32(43), SFInt32(44), SFInt32(44), SFInt32(43), SFInt32(45), SFInt32(44), SFInt32(45), SFInt32(46), SFInt32(46), SFInt32(45), SFInt32(47), SFInt32(46), SFInt32(47), SFInt32(48), SFInt32(48), SFInt32(47), SFInt32(49), SFInt32(48), SFInt32(49), SFInt32(50), SFInt32(50), SFInt32(49), SFInt32(51), SFInt32(50), SFInt32(51), SFInt32(52), SFInt32(52), SFInt32(51), SFInt32(53), SFInt32(52), SFInt32(53), SFInt32(54), SFInt32(54), SFInt32(53), SFInt32(55), SFInt32(54), SFInt32(55), SFInt32(56), SFInt32(56), SFInt32(55), SFInt32(57), SFInt32(56), SFInt32(57), SFInt32(58), SFInt32(58), SFInt32(57), SFInt32(59), SFInt32(58), SFInt32(59), SFInt32(60), SFInt32(60), SFInt32(59), SFInt32(61), SFInt32(60), SFInt32(61), SFInt32(62), SFInt32(62), SFInt32(61), SFInt32(63), SFInt32(62), SFInt32(63), SFInt32(64), SFInt32(64), SFInt32(63), SFInt32(65)]),
                              coord_ : 
                                Coordinate(
                                  point_ : MFVec3f([SFVec3f([2,1.5,0]),SFVec3f([1,1.5,0]),SFVec3f([1.9616,1.5,0.3902]),SFVec3f([0.9808,1.5,0.1951]),SFVec3f([1.8478,1.5,0.7654]),SFVec3f([0.9239,1.5,0.3827]),SFVec3f([1.6629,1.5,1.1111]),SFVec3f([0.8315,1.5,0.5556]),SFVec3f([1.4142,1.5,1.4142]),SFVec3f([0.7071,1.5,0.7071]),SFVec3f([1.1111,1.5,1.6629]),SFVec3f([0.5556,1.5,0.8315]),SFVec3f([0.7654,1.5,1.8478]),SFVec3f([0.3827,1.5,0.9239]),SFVec3f([0.3902,1.5,1.9616]),SFVec3f([0.1951,1.5,0.9808]),SFVec3f([0,1.5,2]),SFVec3f([0,1.5,1]),SFVec3f([-0.3902,1.5,1.9616]),SFVec3f([-0.1951,1.5,0.9808]),SFVec3f([-0.7654,1.5,1.8478]),SFVec3f([-0.3827,1.5,0.9239]),SFVec3f([-1.1111,1.5,1.6629]),SFVec3f([-0.5556,1.5,0.8315]),SFVec3f([-1.4142,1.5,1.4142]),SFVec3f([-0.7071,1.5,0.7071]),SFVec3f([-1.6629,1.5,1.1111]),SFVec3f([-0.8315,1.5,0.5556]),SFVec3f([-1.8478,1.5,0.7654]),SFVec3f([-0.9239,1.5,0.3827]),SFVec3f([-1.9616,1.5,0.3902]),SFVec3f([-0.9808,1.5,0.1951]),SFVec3f([-2,1.5,0]),SFVec3f([-1,1.5,0]),SFVec3f([-1.9616,1.5,-0.3902]),SFVec3f([-0.9808,1.5,-0.1951]),SFVec3f([-1.8478,1.5,-0.7654]),SFVec3f([-0.9239,1.5,-0.3827]),SFVec3f([-1.6629,1.5,-1.1111]),SFVec3f([-0.8315,1.5,-0.5556]),SFVec3f([-1.4142,1.5,-1.4142]),SFVec3f([-0.7071,1.5,-0.7071]),SFVec3f([-1.1111,1.5,-1.6629]),SFVec3f([-0.5556,1.5,-0.8315]),SFVec3f([-0.7654,1.5,-1.8478]),SFVec3f([-0.3827,1.5,-0.9239]),SFVec3f([-0.3902,1.5,-1.9616]),SFVec3f([-0.1951,1.5,-0.9808]),SFVec3f([0,1.5,-2]),SFVec3f([0,1.5,-1]),SFVec3f([0.3902,1.5,-1.9616]),SFVec3f([0.1951,1.5,-0.9808]),SFVec3f([0.7654,1.5,-1.8478]),SFVec3f([0.3827,1.5,-0.9239]),SFVec3f([1.1111,1.5,-1.6629]),SFVec3f([0.5556,1.5,-0.8315]),SFVec3f([1.4142,1.5,-1.4142]),SFVec3f([0.7071,1.5,-0.7071]),SFVec3f([1.6629,1.5,-1.1111]),SFVec3f([0.8315,1.5,-0.5556]),SFVec3f([1.8478,1.5,-0.7654]),SFVec3f([0.9239,1.5,-0.3827]),SFVec3f([1.9616,1.5,-0.3902]),SFVec3f([0.9808,1.5,-0.1951]),SFVec3f([2,1.5,0]),SFVec3f([1,1.5,0])]))))),

                    CADFace(
                      name_ : SFString('bottomCap'),
                      shape_ : 
                        Shape(
                          appearance_ : 
                            Appearance(
                              USE_ : SFString('AppearanceGrey')),
                          geometry_ : 
                            IndexedTriangleSet(
                              index_ : MFInt32([SFInt32(0), SFInt32(1), SFInt32(2), SFInt32(2), SFInt32(1), SFInt32(3), SFInt32(2), SFInt32(3), SFInt32(4), SFInt32(4), SFInt32(3), SFInt32(5), SFInt32(4), SFInt32(5), SFInt32(6), SFInt32(6), SFInt32(5), SFInt32(7), SFInt32(6), SFInt32(7), SFInt32(8), SFInt32(8), SFInt32(7), SFInt32(9), SFInt32(8), SFInt32(9), SFInt32(10), SFInt32(10), SFInt32(9), SFInt32(11), SFInt32(10), SFInt32(11), SFInt32(12), SFInt32(12), SFInt32(11), SFInt32(13), SFInt32(12), SFInt32(13), SFInt32(14), SFInt32(14), SFInt32(13), SFInt32(15), SFInt32(14), SFInt32(15), SFInt32(16), SFInt32(16), SFInt32(15), SFInt32(17), SFInt32(16), SFInt32(17), SFInt32(18), SFInt32(18), SFInt32(17), SFInt32(19), SFInt32(18), SFInt32(19), SFInt32(20), SFInt32(20), SFInt32(19), SFInt32(21), SFInt32(20), SFInt32(21), SFInt32(22), SFInt32(22), SFInt32(21), SFInt32(23), SFInt32(22), SFInt32(23), SFInt32(24), SFInt32(24), SFInt32(23), SFInt32(25), SFInt32(24), SFInt32(25), SFInt32(26), SFInt32(26), SFInt32(25), SFInt32(27), SFInt32(26), SFInt32(27), SFInt32(28), SFInt32(28), SFInt32(27), SFInt32(29), SFInt32(28), SFInt32(29), SFInt32(30), SFInt32(30), SFInt32(29), SFInt32(31), SFInt32(30), SFInt32(31), SFInt32(32), SFInt32(32), SFInt32(31), SFInt32(33), SFInt32(32), SFInt32(33), SFInt32(34), SFInt32(34), SFInt32(33), SFInt32(35), SFInt32(34), SFInt32(35), SFInt32(36), SFInt32(36), SFInt32(35), SFInt32(37), SFInt32(36), SFInt32(37), SFInt32(38), SFInt32(38), SFInt32(37), SFInt32(39), SFInt32(38), SFInt32(39), SFInt32(40), SFInt32(40), SFInt32(39), SFInt32(41), SFInt32(40), SFInt32(41), SFInt32(42), SFInt32(42), SFInt32(41), SFInt32(43), SFInt32(42), SFInt32(43), SFInt32(44), SFInt32(44), SFInt32(43), SFInt32(45), SFInt32(44), SFInt32(45), SFInt32(46), SFInt32(46), SFInt32(45), SFInt32(47), SFInt32(46), SFInt32(47), SFInt32(48), SFInt32(48), SFInt32(47), SFInt32(49), SFInt32(48), SFInt32(49), SFInt32(50), SFInt32(50), SFInt32(49), SFInt32(51), SFInt32(50), SFInt32(51), SFInt32(52), SFInt32(52), SFInt32(51), SFInt32(53), SFInt32(52), SFInt32(53), SFInt32(54), SFInt32(54), SFInt32(53), SFInt32(55), SFInt32(54), SFInt32(55), SFInt32(56), SFInt32(56), SFInt32(55), SFInt32(57), SFInt32(56), SFInt32(57), SFInt32(58), SFInt32(58), SFInt32(57), SFInt32(59), SFInt32(58), SFInt32(59), SFInt32(60), SFInt32(60), SFInt32(59), SFInt32(61), SFInt32(60), SFInt32(61), SFInt32(62), SFInt32(62), SFInt32(61), SFInt32(63), SFInt32(62), SFInt32(63), SFInt32(64), SFInt32(64), SFInt32(63), SFInt32(65)]),
                              coord_ : 
                                Coordinate(
                                  point_ : MFVec3f([SFVec3f([2,-1.5,0]),SFVec3f([1,-1.5,0]),SFVec3f([1.9616,-1.5,-0.3902]),SFVec3f([0.9808,-1.5,-0.1951]),SFVec3f([1.8478,-1.5,-0.7654]),SFVec3f([0.9239,-1.5,-0.3827]),SFVec3f([1.6629,-1.5,-1.1111]),SFVec3f([0.8315,-1.5,-0.5556]),SFVec3f([1.4142,-1.5,-1.4142]),SFVec3f([0.7071,-1.5,-0.7071]),SFVec3f([1.1111,-1.5,-1.6629]),SFVec3f([0.5556,-1.5,-0.8315]),SFVec3f([0.7654,-1.5,-1.8478]),SFVec3f([0.3827,-1.5,-0.9239]),SFVec3f([0.3902,-1.5,-1.9616]),SFVec3f([0.1951,-1.5,-0.9808]),SFVec3f([0,-1.5,-2]),SFVec3f([0,-1.5,-1]),SFVec3f([-0.3902,-1.5,-1.9616]),SFVec3f([-0.1951,-1.5,-0.9808]),SFVec3f([-0.7654,-1.5,-1.8478]),SFVec3f([-0.3827,-1.5,-0.9239]),SFVec3f([-1.1111,-1.5,-1.6629]),SFVec3f([-0.5556,-1.5,-0.8315]),SFVec3f([-1.4142,-1.5,-1.4142]),SFVec3f([-0.7071,-1.5,-0.7071]),SFVec3f([-1.6629,-1.5,-1.1111]),SFVec3f([-0.8315,-1.5,-0.5556]),SFVec3f([-1.8478,-1.5,-0.7654]),SFVec3f([-0.9239,-1.5,-0.3827]),SFVec3f([-1.9616,-1.5,-0.3902]),SFVec3f([-0.9808,-1.5,-0.1951]),SFVec3f([-2,-1.5,0]),SFVec3f([-1,-1.5,0]),SFVec3f([-1.9616,-1.5,0.3902]),SFVec3f([-0.9808,-1.5,0.1951]),SFVec3f([-1.8478,-1.5,0.7654]),SFVec3f([-0.9239,-1.5,0.3827]),SFVec3f([-1.6629,-1.5,1.1111]),SFVec3f([-0.8315,-1.5,0.5556]),SFVec3f([-1.4142,-1.5,1.4142]),SFVec3f([-0.7071,-1.5,0.7071]),SFVec3f([-1.1111,-1.5,1.6629]),SFVec3f([-0.5556,-1.5,0.8315]),SFVec3f([-0.7654,-1.5,1.8478]),SFVec3f([-0.3827,-1.5,0.9239]),SFVec3f([-0.3902,-1.5,1.9616]),SFVec3f([-0.1951,-1.5,0.9808]),SFVec3f([0,-1.5,2]),SFVec3f([0,-1.5,1]),SFVec3f([0.3902,-1.5,1.9616]),SFVec3f([0.1951,-1.5,0.9808]),SFVec3f([0.7654,-1.5,1.8478]),SFVec3f([0.3827,-1.5,0.9239]),SFVec3f([1.1111,-1.5,1.6629]),SFVec3f([0.5556,-1.5,0.8315]),SFVec3f([1.4142,-1.5,1.4142]),SFVec3f([0.7071,-1.5,0.7071]),SFVec3f([1.6629,-1.5,1.1111]),SFVec3f([0.8315,-1.5,0.5556]),SFVec3f([1.8478,-1.5,0.7654]),SFVec3f([0.9239,-1.5,0.3827]),SFVec3f([1.9616,-1.5,0.3902]),SFVec3f([0.9808,-1.5,0.1951]),SFVec3f([2,-1.5,0]),SFVec3f([1,-1.5,0])])))))])])]));
void main() { exit(0); }

File: src/main/vrml/net/coderextreme/data/CADPartChildNoTransformation.mjs
const
   X3D     = require ("x_ite-node"),
   canvas  = X3D .createBrowser (),
   browser = canvas .browser,
   scene   = browser .currentScene;
async function main () {
scene.setProfile(browser.getProfile("CADInterchange"));
scene .addComponent (browser .getComponent ("CADGeometry", 2));
scene .addComponent (browser .getComponent ("EnvironmentalEffects", 1));
scene.addMetaData("title", "CADPartChildNoTransformation.x3d");
scene.addMetaData("description", "Developer scene to explore Mantis Issue 528, providing a baseline example with no transformation of CADPart children.");
scene.addMetaData("creator", "Vince Marchetti, Don Brutzman, Hyokwang Lee, Dick Puk");
scene.addMetaData("translator", "Vince Marchetti bushing generation using python scripts");
scene.addMetaData("reference", "https://www.web3d.org/member-only/mantis/view.php?id=528");
scene.addMetaData("warning", "This scene is intended for specification development only.");
scene.addMetaData("created", "29 June 2012");
scene.addMetaData("modified", "20 October 2019");
scene.addMetaData("identifier", "https://www.web3d.org/x3d/content/examples/Basic/CAD/CADPartChildNoTransformation.x3d");
scene.addMetaData("generator", "X3D-Edit 3.3, https://savage.nps.edu/X3D-Edit");
scene.addMetaData("license", "../license.html");
await browser .loadComponents (scene);
let WorldInfo16 = browser.currentScene.createNode("WorldInfo");
WorldInfo16.title = "CADPartChildNoTransformation.x3d";
browser.currentScene.children = new X3D.MFNode();

browser.currentScene.children[0] = WorldInfo16;

let Background17 = browser.currentScene.createNode("Background");
Background17.skyColor = new X3D.MFColor([0.9607843,1,0.9607843]);
browser.currentScene.children[1] = Background17;

let Viewpoint18 = browser.currentScene.createNode("Viewpoint");
Viewpoint18.description = "Hello CAD bushing";
Viewpoint18.orientation = new X3D.SFRotation([1,0,0,-0.321751]);
Viewpoint18.position = new X3D.SFVec3f([0,5,15]);
browser.currentScene.children[2] = Viewpoint18;

let CADAssembly19 = browser.currentScene.createNode("CADAssembly");
CADAssembly19.name = "DesignPatternAssembly";
let CADPart20 = browser.currentScene.createNode("CADPart");
CADPart20.name = "CADPartExample";
//the rotation and translation attributes serve to position the bushing within the assembly
let CADFace21 = browser.currentScene.createNode("CADFace");
CADFace21.name = "outerSurface";
let Shape22 = browser.currentScene.createNode("Shape");
//note solid='true' and so one-sided external rendering only
let Cylinder23 = browser.currentScene.createNode("Cylinder");
Cylinder23.bottom = False;
Cylinder23.height = 3;
Cylinder23.radius = 2;
Cylinder23.top = False;
geometry = Cylinder23;

let Appearance24 = browser.currentScene.createNode("Appearance");
Appearance24.DEF = "AppearanceGrey";
let Material25 = browser.currentScene.createNode("Material");
material = Material25;

appearance = Appearance24;

shape = Shape22;

CADPart20YYY.children = new X3D.MFNode();

CADPart20ZZZ.children[0] = CADFace21;

let CADFace26 = browser.currentScene.createNode("CADFace");
CADFace26.name = "innerSurface";
let Shape27 = browser.currentScene.createNode("Shape");
//note solid='false' and so two-sided internal + external rendering
let Cylinder28 = browser.currentScene.createNode("Cylinder");
Cylinder28.bottom = False;
Cylinder28.height = 3;
Cylinder28.solid = False;
Cylinder28.top = False;
geometry = Cylinder28;

let Appearance29 = browser.currentScene.createNode("Appearance");
Appearance29.USE = "AppearanceGrey";
appearance = Appearance29;

shape = Shape27;

CADPart20ZZZ.children[1] = CADFace26;

let CADFace30 = browser.currentScene.createNode("CADFace");
CADFace30.name = "topCap";
let Shape31 = browser.currentScene.createNode("Shape");
let IndexedTriangleSet32 = browser.currentScene.createNode("IndexedTriangleSet");
IndexedTriangleSet32.index = new X3D.MFInt32([0,1,2,2,1,3,2,3,4,4,3,5,4,5,6,6,5,7,6,7,8,8,7,9,8,9,10,10,9,11,10,11,12,12,11,13,12,13,14,14,13,15,14,15,16,16,15,17,16,17,18,18,17,19,18,19,20,20,19,21,20,21,22,22,21,23,22,23,24,24,23,25,24,25,26,26,25,27,26,27,28,28,27,29,28,29,30,30,29,31,30,31,32,32,31,33,32,33,34,34,33,35,34,35,36,36,35,37,36,37,38,38,37,39,38,39,40,40,39,41,40,41,42,42,41,43,42,43,44,44,43,45,44,45,46,46,45,47,46,47,48,48,47,49,48,49,50,50,49,51,50,51,52,52,51,53,52,53,54,54,53,55,54,55,56,56,55,57,56,57,58,58,57,59,58,59,60,60,59,61,60,61,62,62,61,63,62,63,64,64,63,65]);
let Coordinate33 = browser.currentScene.createNode("Coordinate");
Coordinate33.point = new X3D.MFVec3f([2,1.5,0,1,1.5,0,1.9616,1.5,0.3902,0.9808,1.5,0.1951,1.8478,1.5,0.7654,0.9239,1.5,0.3827,1.6629,1.5,1.1111,0.8315,1.5,0.5556,1.4142,1.5,1.4142,0.7071,1.5,0.7071,1.1111,1.5,1.6629,0.5556,1.5,0.8315,0.7654,1.5,1.8478,0.3827,1.5,0.9239,0.3902,1.5,1.9616,0.1951,1.5,0.9808,0,1.5,2,0,1.5,1,-0.3902,1.5,1.9616,-0.1951,1.5,0.9808,-0.7654,1.5,1.8478,-0.3827,1.5,0.9239,-1.1111,1.5,1.6629,-0.5556,1.5,0.8315,-1.4142,1.5,1.4142,-0.7071,1.5,0.7071,-1.6629,1.5,1.1111,-0.8315,1.5,0.5556,-1.8478,1.5,0.7654,-0.9239,1.5,0.3827,-1.9616,1.5,0.3902,-0.9808,1.5,0.1951,-2,1.5,0,-1,1.5,0,-1.9616,1.5,-0.3902,-0.9808,1.5,-0.1951,-1.8478,1.5,-0.7654,-0.9239,1.5,-0.3827,-1.6629,1.5,-1.1111,-0.8315,1.5,-0.5556,-1.4142,1.5,-1.4142,-0.7071,1.5,-0.7071,-1.1111,1.5,-1.6629,-0.5556,1.5,-0.8315,-0.7654,1.5,-1.8478,-0.3827,1.5,-0.9239,-0.3902,1.5,-1.9616,-0.1951,1.5,-0.9808,0,1.5,-2,0,1.5,-1,0.3902,1.5,-1.9616,0.1951,1.5,-0.9808,0.7654,1.5,-1.8478,0.3827,1.5,-0.9239,1.1111,1.5,-1.6629,0.5556,1.5,-0.8315,1.4142,1.5,-1.4142,0.7071,1.5,-0.7071,1.6629,1.5,-1.1111,0.8315,1.5,-0.5556,1.8478,1.5,-0.7654,0.9239,1.5,-0.3827,1.9616,1.5,-0.3902,0.9808,1.5,-0.1951,2,1.5,0,1,1.5,0]);
coord = Coordinate33;

geometry = IndexedTriangleSet32;

let Appearance34 = browser.currentScene.createNode("Appearance");
Appearance34.USE = "AppearanceGrey";
appearance = Appearance34;

shape = Shape31;

CADPart20ZZZ.children[2] = CADFace30;

let CADFace35 = browser.currentScene.createNode("CADFace");
CADFace35.name = "bottomCap";
let Shape36 = browser.currentScene.createNode("Shape");
let IndexedTriangleSet37 = browser.currentScene.createNode("IndexedTriangleSet");
IndexedTriangleSet37.index = new X3D.MFInt32([0,1,2,2,1,3,2,3,4,4,3,5,4,5,6,6,5,7,6,7,8,8,7,9,8,9,10,10,9,11,10,11,12,12,11,13,12,13,14,14,13,15,14,15,16,16,15,17,16,17,18,18,17,19,18,19,20,20,19,21,20,21,22,22,21,23,22,23,24,24,23,25,24,25,26,26,25,27,26,27,28,28,27,29,28,29,30,30,29,31,30,31,32,32,31,33,32,33,34,34,33,35,34,35,36,36,35,37,36,37,38,38,37,39,38,39,40,40,39,41,40,41,42,42,41,43,42,43,44,44,43,45,44,45,46,46,45,47,46,47,48,48,47,49,48,49,50,50,49,51,50,51,52,52,51,53,52,53,54,54,53,55,54,55,56,56,55,57,56,57,58,58,57,59,58,59,60,60,59,61,60,61,62,62,61,63,62,63,64,64,63,65]);
let Coordinate38 = browser.currentScene.createNode("Coordinate");
Coordinate38.point = new X3D.MFVec3f([2,-1.5,0,1,-1.5,0,1.9616,-1.5,-0.3902,0.9808,-1.5,-0.1951,1.8478,-1.5,-0.7654,0.9239,-1.5,-0.3827,1.6629,-1.5,-1.1111,0.8315,-1.5,-0.5556,1.4142,-1.5,-1.4142,0.7071,-1.5,-0.7071,1.1111,-1.5,-1.6629,0.5556,-1.5,-0.8315,0.7654,-1.5,-1.8478,0.3827,-1.5,-0.9239,0.3902,-1.5,-1.9616,0.1951,-1.5,-0.9808,0,-1.5,-2,0,-1.5,-1,-0.3902,-1.5,-1.9616,-0.1951,-1.5,-0.9808,-0.7654,-1.5,-1.8478,-0.3827,-1.5,-0.9239,-1.1111,-1.5,-1.6629,-0.5556,-1.5,-0.8315,-1.4142,-1.5,-1.4142,-0.7071,-1.5,-0.7071,-1.6629,-1.5,-1.1111,-0.8315,-1.5,-0.5556,-1.8478,-1.5,-0.7654,-0.9239,-1.5,-0.3827,-1.9616,-1.5,-0.3902,-0.9808,-1.5,-0.1951,-2,-1.5,0,-1,-1.5,0,-1.9616,-1.5,0.3902,-0.9808,-1.5,0.1951,-1.8478,-1.5,0.7654,-0.9239,-1.5,0.3827,-1.6629,-1.5,1.1111,-0.8315,-1.5,0.5556,-1.4142,-1.5,1.4142,-0.7071,-1.5,0.7071,-1.1111,-1.5,1.6629,-0.5556,-1.5,0.8315,-0.7654,-1.5,1.8478,-0.3827,-1.5,0.9239,-0.3902,-1.5,1.9616,-0.1951,-1.5,0.9808,0,-1.5,2,0,-1.5,1,0.3902,-1.5,1.9616,0.1951,-1.5,0.9808,0.7654,-1.5,1.8478,0.3827,-1.5,0.9239,1.1111,-1.5,1.6629,0.5556,-1.5,0.8315,1.4142,-1.5,1.4142,0.7071,-1.5,0.7071,1.6629,-1.5,1.1111,0.8315,-1.5,0.5556,1.8478,-1.5,0.7654,0.9239,-1.5,0.3827,1.9616,-1.5,0.3902,0.9808,-1.5,0.1951,2,-1.5,0,1,-1.5,0]);
coord = Coordinate38;

geometry = IndexedTriangleSet37;

let Appearance39 = browser.currentScene.createNode("Appearance");
Appearance39.USE = "AppearanceGrey";
appearance = Appearance39;

shape = Shape36;

CADPart20ZZZ.children[3] = CADFace35;

CADAssembly19YYY.children = new X3D.MFNode();

CADAssembly19ZZZ.children[0] = CADPart20;

browser.currentScene.children[3] = CADAssembly19;

}
main ();
